OpenRA.Mods.Cnc/Traits/Disguise.cs

```diff
@@ -243,5 +251,45 @@ void INotifyInfiltration.Infiltrating(Actor self)
 			if (info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Infiltrate))
 				DisguiseAs(null);
 		}
+
+		void ITick.Tick(Actor self)
+		{
+			if (info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Move) && lastPos != null && lastPos.Value != self.Location)
```

Comment:
I am not a fan of this approach. Isn't this exactly why we have `IMove.IsMoving`?
Also this will trigger a drop of disguise on teleport or similar force-move, which might be an unwanted side effect, while `IsMoving` won't.

Reply:
This is consistent with what Cloak's doing. https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/Cloak.cs#L150

Reply:
Then I'm not a fan of that one either :D
My question stands: can't we use `IMove.IsMoving` and if we can, why aren't we?

Reply:
`IMove.IsMoving` has never worked properly.  It has recently-ish been improved, but at least #11885 is still a problem.